pandas/core/indexes/period.py

```diff
@@ -906,6 +923,10 @@ def base(self):
                       FutureWarning, stacklevel=2)
         return np.asarray(self._data)
 
+    def memory_usage(self, deep=False):
+        result = super(PeriodIndex, self).memory_usage(deep=deep)
+        result += self._int64index.memory_usage(deep=deep)
```

Comment:
this only should add the int64index if it’s actually created

Reply:
Checking for `'_int64engine'` in `self._cache` now, which avoids triggering creation.